From fdb02ee44ed364b140a1808f97426a274ca289c7 Mon Sep 17 00:00:00 2001
From: Xie XiuQi <xiexiuqi@huawei.com>
Date: Fri, 26 Feb 2021 20:21:15 +0800
Subject: [PATCH] arm64/mpam: don't allowd create mon_groups when out of
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: fdb02ee44ed364b140a1808f97426a274ca289c7

 mon/pmg

hulk inclusion
category: feature
feature: ARM MPAM support
bugzilla: 48265
CVE: NA

--------------------------------

Signed-off-by: Xie XiuQi <xiexiuqi@huawei.com>
Reviewed-by: Hanjun Guo <guohanjun@huawei.com>
Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
Signed-off-by: Wang ShaoBo <bobo.shaobowang@huawei.com>
Reviewed-by: Cheng Jian <cj.chengjian@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/resctrlfs.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/fs/resctrlfs.c b/fs/resctrlfs.c
index d75004405803..31ecedf433ec 100644
--- a/fs/resctrlfs.c
+++ b/fs/resctrlfs.c
@@ -596,7 +596,12 @@ static int mkdir_resctrl_prepare(struct kernfs_node *parent_kn,
 
 	if (resctrl_mon_capable) {
 #ifdef CONFIG_ARM64
-		resctrl_mkdir_ctrlmon_mondata(kn, rdtgrp, &rdtgrp->mon.mon_data_kn);
+		ret = resctrl_mkdir_ctrlmon_mondata(kn, rdtgrp, &rdtgrp->mon.mon_data_kn);
+		if (ret < 0) {
+			rdt_last_cmd_puts("out of monitors or PMGs\n");
+			goto out_destroy;
+		}
+
 #else
 		ret = alloc_mon_id();
 		if (ret < 0) {
-- 
2.26.2

